test-harness/src/main/java/io/jenkins/plugins/casc/misc/junit/jupiter/WithJenkinsConfiguredWithCode.java

```diff
@@ -0,0 +1,90 @@
+package io.jenkins.plugins.casc.misc.junit.jupiter;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ * JUnit 5 meta annotation providing {@link JenkinsRule JenkinsRule} integration.
+ *
+ * <p>Test methods using the rule extension need to accept it by {@link JenkinsRule JenkinsRule} parameter;
+ * each test case gets a new rule object.
+ * An annotated method without a {@link JenkinsRule JenkinsRule} parameter behaves as if it were not annotated.
+ *
+ * <p>Annotating a <em>class</em> provides access for all of its tests.
+ * Unrelated test cases can omit the parameter.
+ *
+ * <blockquote>
+ *
+ * <pre>
+ * &#64;WithJenkinsConfiguredWithCode
+ * class ExampleJUnit5Test {
+ *
+ *     &#64;Test
+ *     public void example(JenkinsRule r) {
+ *         // use 'r' ...
+ *     }
+ *
+ *     &#64;Test
+ *     public void exampleNotUsingRule() {
+ *         // ...
+ *     }
+ * }
+ * </pre>
+ *
+ * </blockquote>
+ *
+ * <p>Annotating a <i>method</i> limits access to the method.
+ *
+ * <blockquote>
+ *
+ * <pre>
+ * class ExampleJUnit5Test {
+ *
+ *     &#64;WithJenkinsConfiguredWithCode
+ *     &#64;Test
+ *     public void example(JenkinsRule r) {
```

Comment:
```suggestion
 *     public void example(JenkinsConfiguredWithCodeRule r) {
```

Reply:
I wasn't aware of any difference in the `JenkinsConfiguredWithCodeRule` that necessitated it being injected. Is there additional, useful API to the test writers?

Reply:
no but it's what you injected in the example test, the javadoc should match what's being tested.
https://github.com/bstopp/configuration-as-code-plugin/blob/feat/junit-jupiter/test-harness/src/test/java/io/jenkins/plugins/casc/junit/jupiter/JenkinsConfiguredWithCodeMethodRuleTest.java